### ISSUE TITLE
508-defect-2: Destructive action with no confirmation step or screen reader feedback

### ISSUE THREAD

**Issue by jeana-adhoc** [department-of-veterans-affairs/va.gov-team#59193]
### Point of contact

@jeana-adhoc

### Severity level

2, Serious. Should be fixed in 1-2 sprints post-launch.

### Details

On "Step 7 of 12: Treatment records", after a user enters one treatment record, and selects to enter another, a "Remove" button appears. The Remove button is a single-click destructive action. When clicking Remove, the item is removed without any user confirmation. Additionally, there's nothing announced by screen readers to indicate that a change has occurred --- focus just moves to the next focusable element.

It's easy to imagine some scenarios where a user might accidentally trigger the Remove button and not realize that they did it. Screen reader users may hit the wrong key but think they hit [Tab]. A low-vision user on a touch device might pinch-to-zoom and accidentally tap the Remove button in the process, and not see the issue disappear.

Since users get an opportunity to review everything prior to submission, this is probably not a violation of experience standard 03.09 or [WCAG 3.3.4](https://www.w3.org/WAI/WCAG21/Understanding/error-prevention-legal-financial-data.html). But I do think it's in the spirit of WCAG 3.3.4.  

### Reproduction steps

1. Add a facility in step 7
2. Click "Add another treatment facility"
3. Scroll down and click "Remove" on the new facility
4. No alert/notice is given as to the removal of a facility

### Proposed solution or next steps

The [sketch file](https://sketch.com/s/c8c31a8a-19ae-4cbc-81ef-40eb8f4f8c52) has been updated to reflect this. And I've attached a screenshot here:
<img width="500" alt="image" src="https://github.com/department-of-veterans-affairs/va.gov-team/assets/117098918/7a14dcae-d6aa-44fa-a861-1eebbd4be86c">

This should use the [VA Modal - Warning component](https://design.va.gov/storybook/?path=/docs/components-va-modal--warning). This should also be available when "remove" might be clicked from the review page.

**note:** We can't pass in the specific facility name into the modal.

### References, articles, or WCAG support

1. [WCAG 3.3.4](https://www.w3.org/WAI/WCAG21/Understanding/error-prevention-legal-financial-data.html).
2. This ticket is about the same issue: https://github.com/department-of-veterans-affairs/va.gov-team/issues/56082
3. [VA Experience Standards](https://depo-platform-documentation.scrollhelp.site/collaboration-cycle/va-gov-experience-standards) 03.09
4. [PR For supplemental claims to add similar modal](https://github.com/department-of-veterans-affairs/vets-website/pull/23924)

### Type of issue

- [X] Screenreader
- [X] Keyboard
- [X] Focus
- [ ] Headings
- [ ] Color
- [ ] Zoom
- [ ] Semantics
- [ ] Axe-core
- [ ] Something else

**Comment by gmrabian** [comment]
Added Nick just because this might end up being implemented on the platform side.

**Comment by ndsprinkle** [comment]
https://vfs.atlassian.net/wiki/spaces/pilot/pages/2104623233/VA+Forms+Library+-+About+schema+and+uiSchema#ui-options

I added documentation to the bottom of the `ui:options` stuff. This may change in the future if we make this the default behavior. We also may add additional docs for array stuff in general based on our experience with 4142's array and include the confirmation modal in that.
